uaclient/apt.py

```diff
@@ -214,6 +215,29 @@ def get_apt_cache_policy(
     )
 
 
+class PreserveAptCfg:
+    def __init__(self, apt_func):
+        self.apt_func = apt_func
+        self.current_apt_cfg = {}  # Dict[str, Any]
+
+    def __enter__(self):
+        apt_pkg.init_config()
+        cfg = apt_pkg.config
+        self.current_apt_cfg = {
+            key: copy.deepcopy(cfg.get(key)) for key in cfg.keys()
+        }
+
+        return self.apt_func()
+
+    def __exit__(self, type, value, traceback):
+        cfg = apt_pkg.config
+        # We need to restore the apt cache configuration after creating our
+        # cache, otherwise we may break people interacting with the
+        # library after importing our modules.
+        for key in self.current_apt_cfg.keys():
+            cfg.set(key, self.current_apt_cfg[key])
```

Comment:
If I am not mistaken we can simplify this by just assigning the config back to apt_pkg.
```suggestion
        apt_pkg.config = self.current_apt_cfg
```